third_party/nvfuser/test/test_gpu1.cpp

```diff
@@ -7715,6 +7715,83 @@ TEST_F(NVFuserTest, FusionMagicSchedulerLayerNormalization_CUDA) {
       "");
 }
 
+TEST_F(
+    NVFuserTest,
+    FusionPersistentBufferProjectionAfterWelfordTranslate_CUDA) {
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+  const float kEps = 1e-5;
+  Double* eps_ptr = IrBuilder::create<Double>(kEps);
+
+  DataType dtype = DataType::Half;
+  constexpr int64_t dim0 = 2048;
+  constexpr int64_t dim1 = 10240;
+  std::vector<int64_t> input_shape{dim0, dim1};
+  std::vector<int64_t> norm_shape{dim1};
+  auto input_half = makeContigTensor(2, dtype);
+  auto weight_half = makeContigTensor(1, dtype);
+  auto bias_half = makeContigTensor(1, dtype);
+  fusion.addInput(input_half);
+  fusion.addInput(weight_half);
+  fusion.addInput(bias_half);
+  auto input = castOp(DataType::Float, input_half);
+  auto weight = castOp(DataType::Float, weight_half);
+  auto bias = castOp(DataType::Float, bias_half);
+  auto result = layer_norm(input, norm_shape, weight, bias, eps_ptr);
+  auto result_output = castOp(dtype, result.output);
+  fusion.addOutput(result_output);
+  fusion.addOutput(result.mean);
+  fusion.addOutput(result.invstd);
+
+  auto options =
+      at::TensorOptions().dtype(data_type_to_aten(dtype)).device(at::kCUDA, 0);
+  at::Tensor aten_input = at::randn(input_shape, options);
+  c10::optional<at::Tensor> aten_weight = at::randn({input_shape[1]}, options);
+  c10::optional<at::Tensor> aten_bias = at::randn({input_shape[1]}, options);
+  auto aten_outputs = at::native_layer_norm(
+      aten_input, norm_shape, aten_weight, aten_bias, kEps);
+
+  // welford translate
+  KernelArgumentHolder runtime_inputs =
+      KernelArgumentHolder::createKernelArgumentHolder(
+          {aten_input, aten_weight, aten_bias});
+  bool isTranslated =
+      SegmentCandidateFinder::TranslateWelfordInFusion(&fusion, runtime_inputs);
+  TORCH_INTERNAL_ASSERT(isTranslated);
+
+  // project persistent buffer to input
+  reduction_scheduler_utils::projectPersistentBuffers(&fusion);
```

Comment:
No need to do this, and actually I'm not sure if it should work. `projectPersistentBuffers` may add dummy outputs, which are subsequently removed when used in schedulers, but here they seem to remain, which may result in unexpected outputs.

Instead, I believe you can just assert that :

```
auto persistent_buffer_info = scheduler_utils::persistentBuffers(&fusion);
persistent_buffer_info.projectable_buffers().size() == 1
persistent_buffer_info.projectable_buffer_inputs().size() == 1
persistent_buffer_info.projectable_buffer_inputs()[0] == input_half
```